src/arch/riscv/traps.S

```diff
@@ -98,50 +98,43 @@ trap_entry:
   csrr x1,  sepc
   STORE   x1, (33*REGBYTES)(t0)
 
-#ifdef CONFIG_FASTPATH
-  li t3, SYSCALL_REPLY_RECV
-  beq a7, t3, fp_reply_recv
-
-  li t3, SYSCALL_CALL
-  beq a7, t3, fp_call
-#endif
-
   /* Check if it's an interrupt */
   bltz s0, interrupt
 
-  li   s4, 8       /* ratified priv has value 8 for ecall exception */
+  /* ratified priv has value 8 for ecall exception */
+  li   s4, 8
   bne  s0, s4, exception
 
-syscall:
+  /* If it's not an exception or an interrupt it's a syscall */
   /* Set the return address to sepc + 4 in the case of a system/environment call */
   addi x1, x1, 4
   /* Save NextIP */
   STORE   x1, (34*REGBYTES)(t0)
 
+#ifdef CONFIG_FASTPATH
+  li t3, SYSCALL_REPLY_RECV
+  beq a7, t3, fp_reply_recv
+
+  li t3, SYSCALL_CALL
+  beq a7, t3, c_handle_fastpath_call
+#endif
+
+syscall:
+
+  /* move syscall number to 3rd argument */
   mv a2, a7
 
   j c_handle_syscall
 
 fp_reply_recv:
 
-  addi x1, x1, 4
```

Comment:
What is the commit policy actually concerning trivial changes, should removing this be a separate commit?